apps/android/app/src/main/java/chat/simplex/app/views/database/ChatArchiveView.kt

```diff
@@ -0,0 +1,147 @@
+  val chatArchiveTime = remember { mutableStateOf(m.controller.appPrefs.chatArchiveTime.get()) }
+  val saveArchiveLauncher = rememberSaveArchiveLauncher(cxt = context, archivePath)
+  ChatArchiveLayout(
+    title,
+    chatArchiveTime,
+    saveArchive = { saveArchiveLauncher.launch(archivePath.substringAfterLast("/")) },
+    deleteArchiveAlert = { deleteArchiveAlert(m, archivePath, chatArchiveName, chatArchiveTime) }
+  )
+}
+
+@Composable
+fun ChatArchiveLayout(
+  title: String,
+  chatArchiveTime: MutableState<Instant?>,
+  saveArchive: () -> Unit,
+  deleteArchiveAlert: () -> Unit
+) {
+  Column(
+    Modifier.fillMaxWidth(),
+    horizontalAlignment = Alignment.Start,
+  ) {
+    @Composable fun divider() = Divider(Modifier.padding(horizontal = 8.dp))
+    Text(
+      title,
+      Modifier.padding(start = 16.dp, bottom = 24.dp),
+      style = MaterialTheme.typography.h1
+    )
+
+    SettingsSectionView(stringResource(R.string.chat_database_section)) {
+      SettingsActionItem(
+        Icons.Outlined.IosShare,
+        stringResource(R.string.save_archive),
+        saveArchive,
+        textColor = MaterialTheme.colors.primary
+      )
+      divider()
+      SettingsActionItem(
+        Icons.Outlined.Delete,
+        stringResource(R.string.delete_archive),
+        deleteArchiveAlert,
+        textColor = Color.Red
+      )
+    }
+    val chatArchiveTimeVal = chatArchiveTime.value
+    if (chatArchiveTimeVal != null) {
+      val archiveTs = SimpleDateFormat("yyyy-MM-dd HH:mm:ss Z", Locale.US).format(Date.from(chatArchiveTimeVal.toJavaInstant()))
+      SettingsSectionFooter(
+        String.format(generalGetString(R.string.archive_created_on_ts), archiveTs)
+      )
+    }
+  }
+}
+
+@Composable
+private fun rememberSaveArchiveLauncher(cxt: Context, chatArchivePath: String): ManagedActivityResultLauncher<String, Uri?> =
+  rememberLauncherForActivityResult(
+    contract = ActivityResultContracts.CreateDocument(),
+    onResult = { destination ->
+      destination?.let {
+        val contentResolver = cxt.contentResolver
+        contentResolver.openOutputStream(destination)?.let { stream ->
+          val outputStream = BufferedOutputStream(stream)
+          val file = File(chatArchivePath)
+          outputStream.write(file.readBytes())
+          outputStream.close()
+          Toast.makeText(cxt, generalGetString(R.string.file_saved), Toast.LENGTH_SHORT).show()
+        }
+      }
+    }
+  )
+
+private fun deleteArchiveAlert(m: ChatModel, archivePath: String, chatArchiveName: MutableState<String?>, chatArchiveTime: MutableState<Instant?>) {
+  AlertManager.shared.showAlertDialog(
+    title = generalGetString(R.string.delete_chat_archive_question),
+    confirmText = generalGetString(R.string.delete_verb),
+    onConfirm = {
+      val fileDeleted = File(archivePath).delete()
+      if (fileDeleted) {
+        m.controller.appPrefs.chatArchiveName.set(null)
+        chatArchiveName.value = null
```

Comment:
deleting it should close the page, there is no need to update these states (and no need for them to be mutable)